### PR TITLE
Update for nextcloud 25

### PR DESCRIPTION
**Pull request by mwinkens**

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
+	width: 100%;
+	table-layout: fixed /* fix a bug on firefox*/
 }
 
 .b2sharebridge .publish-queue-list th {
-    /*text-align: left;*/
+	/*text-align: left;*/
+}
+
+.icon-b2share:after{
```
Comment (mwinkens) [inline css/style.css]: @mwinkens remove this, this didn't work

File: lib/Events/LoadSidebarListener.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace OCA\B2shareBridge\Events;
+
+use OCP\Util;
+use OCP\EventDispatcher\Event;
+use OCA\B2shareBridge\AppInfo\Application;
+use OCP\EventDispatcher\IEventListener;
+use OCA\Files\Event\LoadSidebar;
+
+class LoadSidebarListener implements IEventListener
```
Comment (mwinkens) [inline lib/Events/LoadSidebarListener.php]: @mwinkens remove this file, it does nothing!

File: lib/Model/DepositStatus.php [hunk 3]
```diff
@@ -86,4 +86,21 @@ public function getHyperlink()
 
         return urldecode($result);
     }
+
+    public function toJson()
```
Comment (mwinkens) [inline lib/Model/DepositStatus.php]: @mwinkens maybe add php docs

File: lib/Publish/Ipublish.php [hunk 4]
```diff
@@ -14,9 +14,11 @@
```
Comment (mwinkens) [inline lib/Publish/Ipublish.php:14]: @mwinkens maybe you should refactor this interface in a later PR

File: package.json [hunk 5]
```diff
@@ -1,8 +1,55 @@
...
-    "jquery": "^3.6.0"
+    "@mdi/js": "^7.1.96",
+    "@nextcloud/axios": "^2.3.0",
+    "@nextcloud/dialogs": "^3.2.0",
+    "@nextcloud/files": "^2.1.0",
+    "@nextcloud/router": "^2.0.1",
+    "@nextcloud/typings": "^1.6.0",
+    "@nextcloud/vue": "^7.4.0",
+    "bootstrap-vue": "^2.23.1",
+    "jquery": "^3.6.0",
```
Comment (mwinkens) [inline package.json:32]: @mwinkens TODO: get rid of this in a future PR

File: src/main.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
```
Comment (mwinkens) [inline src/main.js]: @mwinkens remove this copyright, this is not a file from John!

File: tests/unit/Controller/ViewControllerTest.php [hunk 7]
```diff
@@ -10,30 +10,42 @@
...
     public function setUp(): void
     {
-        $request = $this->getMockBuilder('OCP\IRequest')->getMock();
+        parent::setUp();
+        $this->request = $this->getMockBuilder('OCP\IRequest')->getMock();
         $config = $this->getMockBuilder('OCP\IConfig')->getMock();
-        $deposit_mapper = $this->getMockBuilder('OCA\B2shareBridge\Model\DepositStatusMapper')
-            ->setMethods(['findAllForUser','findAllForUserAndStateString'])
+        $this->deposit_mapper = $this->getMockBuilder('OCA\B2shareBridge\Model\DepositStatusMapper')
+            ->setMethods(['findAllForUser', 'findAllForUserAndStateString', 'findLastUpdate'])
```
Comment (mwinkens) [inline tests/unit/Controller/ViewControllerTest.php]: @mwinkens remove "findLastUpdate" as it exists no more

**Review by mwinkens — COMMENTED** [review]
stuff to do

**Comment by apw1388** [comment]
Wouldn't that fundamental change lead to a new mayor version?

**Comment by mwinkens** [comment]
> Wouldn't that fundamental change lead to a new mayor version?

Since I changed api endpoints, you are right!

File: lib/Publish/Ipublish.php (hunk 4 above)
Comment (mwinkens) [reply lib/Publish/Ipublish.php:14]: done in #70

**Pull request merged**